frontend/src/app/services/resource.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpClient,
+  HttpErrorResponse,
+  HttpParams,
+} from '@angular/common/http';
+import { environment } from '../../environments/environment';
+import { Observable, throwError } from 'rxjs';
+import { catchError, map, switchMap, take } from 'rxjs/operators';
+import { IResponse } from '../core/models/response';
+
+@Injectable({
+  providedIn: 'root',
+})
+export abstract class ResourceService<T> {
+  private readonly APIUrl = environment.apiUrl + this.getResourceUrl();
+
+  constructor(protected httpClient: HttpClient) {}
+
+  abstract getResourceUrl(): string;
+
+  toServerModel(entity: T): string {
+    return JSON.stringify(entity);
+  }
+
+  fromServerModel(json: IResponse): T[] {
+    let item = JSON.parse(json.data);
+    var data: T[] = Array.isArray(item) ? item : [item];
+    return data;
+  }
+
+  getList(index: number, page: number): Observable<T[]> {
+    let params = new HttpParams()
+      .set('limit', index.toString())
+      .set('offset', page.toString());
+
+    return this.httpClient
+      .get<IResponse[]>(`/${this.APIUrl}?${params.toString()}`)
+      .pipe(
+        switchMap((list) => list.map((item) => this.fromServerModel(item))),
+        catchError(this.handleError)
+      );
+  }
+
+  get(id: string | number): Observable<T> {
```

Comment:
I don't think we need two different methods for getting a single resource or a list. Make it just one get